xmppserver/src/main/java/org/jivesoftware/openfire/disco/IQDiscoInfoHandler.java

```diff
@@ -513,6 +514,46 @@ public static DataForm getFirstDataForm(Set<DataForm> dataForms){
         return  dataForms.stream().filter(Objects::nonNull).findAny().orElse(null);
     }
 
+    /**
+     * Set all Software Version data  
+     * responsed by the peer for the Software information request Service Discovery (XEP-0232)
+     * @param Element query represented on the response of the peer
+     * @param LocalSession session represented the LocalSession with peer
+     */
+    public static void setSoftwareVersionDataFormFromDiscoInfo(Element query ,LocalSession session){
+        Boolean containDisco = new Boolean("false");
+        Boolean typeformDataSoftwareInfo = new Boolean("false");
+        List<Element> elements = query.elements();
+        if (elements.isEmpty() && elements!= null && elements.size()>0 && session != null){
+            for (Element element : elements){
```

Comment:
You need to make sure that the `elements` collection is not null in a for-each loop like this:

    for (Object element : elements )

It is perfectly OK to use a collection that is empty (the loop will simply be skipped).